src/shared/sam/cli/samCliLocalInvoke.ts

```diff
@@ -208,7 +208,7 @@ export interface SamCliLocalInvokeInvocationArguments {
  * Yet another `sam` CLI wrapper.
  */
 export class SamCliLocalInvokeInvocation {
-    private readonly config = new SamCliSettings()
+    private readonly config = SamCliSettings.instance
```

Comment:
Using a singleton isn't really required. Just makes the code a lot more clear compared to a chain of default initializers. It's also slightly easier to test as we can stub `instance` instead of the constructor. Of course, the preferred way to test would be to make it a parameter. But that's not always feasible, especially with legacy code.

In TS 4.7, I should be able to make this modular singleton pattern a lot more ergonomic. It has some very exciting features regarding type instantiation.